distribution/xfer/download.go

```diff
@@ -33,11 +36,12 @@ func (ldm *LayerDownloadManager) SetConcurrency(concurrency int) {
 }
 
 // NewLayerDownloadManager returns a new LayerDownloadManager.
-func NewLayerDownloadManager(layerStore layer.Store, concurrencyLimit int, options ...func(*LayerDownloadManager)) *LayerDownloadManager {
+func NewLayerDownloadManager(layerStore layer.Store, concurrencyLimit int, cacheRoot string, options ...func(*LayerDownloadManager)) *LayerDownloadManager {
```

Comment:
cacheRoot is in the constructor of cachedir Manager and concurrentyLimit is in the constructor of TransferManager. They are not actually optional right now and making them optional seems unnecessary at this time IMO.